packages/frontend/src/components/nft/NFTDetailWrapper.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import { selectAccountId, selectBalance } from "../../redux/slices/account";
+import {
+    actions as nftActions,
+    selectTokenForAccountForContractForTokenId,
+    selectTransferredTokenForContractForTokenId,
+} from '../../redux/slices/nft';
+import { NFTDetail } from "./NFTDetail";
+
+export function NFTDetailWrapper ({
+    match,
+    history
+}) {
+    const { contractId: contractName, tokenId } = match.params;
+    const accountId = useSelector(selectAccountId);
+    const { balanceAvailable: nearBalance} = useSelector(selectBalance);
+    const nft = useSelector((state) => selectTokenForAccountForContractForTokenId(state, {
+        accountId,
+        contractName,
+        tokenId,
+    }));
+
+    const transferredNft = useSelector((state) => selectTransferredTokenForContractForTokenId(state, {
+        contractName,
+        tokenId,
+    }));
+
+    const dispatch = useDispatch();
+    const { fetchNFTs } = nftActions;
+
+    useEffect(() => {
+        if (accountId && !nft) {
+            dispatch(fetchNFTs({ accountId, contractName }));
```

Comment:
This should only happen if the user refreshes the page but it's still probably worth adding an action to fetch just the token with `tokenId` rather than fetching all for this `accountId` + `contractName` pair.

Reply:
Added new action